### PR TITLE
refactor: Reduce executor cache interface to one method

### PR DESCRIPTION
**Pull request by TrevorBergeron**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/session/executor.py [hunk 1]
```diff
@@ -108,6 +110,7 @@ def __init__(
         ] = weakref.WeakKeyDictionary()
         self.metrics = metrics
         self.bqstoragereadclient = bqstoragereadclient
+        self.nondeterminism_only_once = (nondeterminism_only_once,)
```
Comment (Genesis929) [inline bigframes/session/executor.py:113]: self.nondeterminism_only_once seems not used?

Comment (TrevorBergeron) [reply bigframes/session/executor.py:113]: sorry, leaked from some other changes I was working on. removed from new revision

Comment (Genesis929) [reply bigframes/session/executor.py:113]: I don't see new commit, maybe you didn't push?

Comment (TrevorBergeron) [reply bigframes/session/executor.py]: didn't actually push, now its pushed

**Review by Genesis929 — APPROVED** [review]

**Pull request merged**